contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,906 @@
+        }
+        finalizationBatchLimit = _finalizationBatchLimit;
+        emit UpdatedFinalizationBatchLimit(finalizationBatchLimit);
+    }
+
+    /**
+     * @notice updates the undelegation period of a withdraw request
+     * @dev only `DEFAULT_ADMIN_ROLE` role can call
+     * @param _undelegationPeriodInBlocks new value of undelegationPeriodInBlocks
+     */
+    function updateUndelegationPeriodInBlocks(uint256 _undelegationPeriodInBlocks)
+        external
+        override
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        if (_undelegationPeriodInBlocks <= finalizationBatchLimit) {
+            revert InvalidInput();
+        }
+        undelegationPeriodInBlocks = _undelegationPeriodInBlocks;
+        emit UpdatedUndelegationPeriodInBlocks(undelegationPeriodInBlocks);
+    }
+
+    /**
+     * @notice updates the value of minimum block delay to finalize withdraw requests
+     * @dev only `DEFAULT_ADMIN_ROLE` role can call
+     * @param _minBlockDelayToFinalizeRequest new value of minBlockDelayToFinalizeRequest
+     */
+    function updateMinBlockDelayToFinalizeRequest(uint256 _minBlockDelayToFinalizeRequest)
+        external
+        override
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        minBlockDelayToFinalizeRequest = _minBlockDelayToFinalizeRequest;
+        emit UpdatedMinBlockDelayToFinalizeRequest(minBlockDelayToFinalizeRequest);
+    }
+
+    /**
+     * @notice updates the value of `maxNonRedeemedDelegatorRequestCount`
+     * @dev only `MANAGER` role can call
+     * @param _count new count of maxNonRedeemedDelegatorRequest
+     */
+    function updateMaxNonRedeemedDelegatorRequestCount(uint256 _count) external override {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        maxNonRedeemedDelegatorRequestCount = _count;
+        emit UpdatedMaxNonRedeemedDelegatorRequestCount(_count);
+    }
+
+    /// @notice update the address of staderConfig
+    function updateStaderConfig(address _staderConfig) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        UtilLib.checkNonZeroAddress(_staderConfig);
+        staderConfig = IStaderConfig(_staderConfig);
+        emit UpdatedStaderConfig(_staderConfig);
+    }
+
+    //Getters
+
+    /// @notice return the list of ongoing withdraw requestIds for a user
+    function getRequestIdsByDelegator(address _delegator) external view override returns (uint256[] memory) {
+        return requestIdsByDelegatorAddress[_delegator];
+    }
+
+    /**
+     * @notice Return the utilized balance of account based on stored data
+     * @param account The address whose balance should be calculated
+     * @return The calculated balance
+     */
+    function utilizerBalanceStored(address account) external view override returns (uint256) {
+        return _utilizerBalanceStoredInternal(account);
+    }
+
+    /// @notice Calculates the current delegation rate per block
+    function getDelegationRate() external view override returns (uint256) {
```

Comment:
Would call this fn `getDelegationRatePerBlock` to make it explicit that it isn't the "rate"